tests/test_interactions.py

```diff
@@ -0,0 +1,153 @@
+            :param delta_y: how much we want to drag the node
+            :param dir_name: The directory in which we store our screenshots
+            :return: the difference between the position after drag and the starting position
+            """
+            actions.reset_actions()
+            actions.move_to_element_with_offset(
+                self.driver.find_element_by_tag_name('body'), x, y
+            )
+            actions.drag_and_drop_by_offset(source=None, xoffset=delta_x, yoffset=delta_y)
+            actions.click()
+            actions.perform()
+            time.sleep(1)
+
+            elem_json = json.loads(elem.text)
+            new_pos = elem_json.get('renderedPosition')
+            clicked_label = elem_json.get('data', {}).get('label')
+
+            diff_x = round(new_pos['x'] - x)
+            diff_y = round(new_pos['y'] - y)
+
+            save_screenshot(
+                dir_name,
+                'Dragged{}By{}x{}y'.format(clicked_label.replace(' ', ''), diff_x, diff_y)
+            )
+
+            return diff_x, diff_y
+
+        def perform_clicking(x, y, elem, dir_name='interactions'):
+            """
+            :param x: The position on the screen where we want to click
+            :param y: The position on the screen where we want to click
+            :param elem: The element object from where we retrieve the JSON
+            :param dir_name: The directory in which we store our screenshots
+            :return: The label of element most recently clicked, if any
+            """
+            actions.reset_actions()
+            actions.move_to_element_with_offset(
+                self.driver.find_element_by_tag_name('body'), x, y
+            )
+            actions.click()
+            actions.perform()
+
+            time.sleep(1)
+            clicked_label = json.loads(elem.text).get('data', {}).get('label')
+
+            save_screenshot(dir_name, 'Clicked' + clicked_label.replace(' ', ''))
+
+            return clicked_label
+
+        def perform_mouseover(x, y, elem, dir_name='interactions'):
+            actions.reset_actions()
+            actions.move_to_element_with_offset(
+                self.driver.find_element_by_tag_name('body'), x - 100, y
+            )
+            actions.move_by_offset(100, 0)
+            actions.perform()
+            time.sleep(1)
+
+            mouseover_label = json.loads(elem.text).get('label')
+
+            save_screenshot(dir_name, 'Mouseover' + mouseover_label.replace(' ', ''))
+
+            return mouseover_label
+
+        drag_error = "Unable to drag Cytoscape nodes properly"
+        click_error = "Unable to click Cytoscape nodes properly"
+        mouseover_error = "Unable to mouseover Cytoscape nodes properly"
+
+        init_pos = {
+            'Node 1': (80.94611044209678, 333.54879281525285),
+            'Node 2': (375.64032747402433, 628.2430098471805),
+            'Node 3': (277.40892179671516, 514.2945792615018 - 20),
```

Comment:
where is this `- 20` coming from?

Reply:
I had to add an offset since that "Node 6" is overlapping with "Node 3" at that particular position.

Reply:
cool - maybe add a comment?

Reply:
Will do.